// routeros-api/dist/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dashedCaseToSnakeCase = exports.dashedCaseToCamelCase = exports.camelCaseOrSnakeCaseToDashedCase = exports.lookForIdParameterAndReturnItsValue = exports.debounce = void 0;
exports.debounce = (callback, timeout = 0) => {
    let timeoutObj = null;
    return {
        run: (...args) => {
            const context = this;
            clearTimeout(timeoutObj);
            timeoutObj = setTimeout(() => callback.apply(context, args), timeout);
        },
        cancel: () => {
            clearTimeout(timeoutObj);
        },
    };
};
/**
 * For for any .id search on the query and return it
 *
 * @param queries array of transformed query
 */
function lookForIdParameterAndReturnItsValue(queries) {
    let val = null;
    for (const query of queries) {
        if (query.includes("numbers=") || query.includes(".id=")) {
            val = query.split("=").pop();
        }
    }
    return val;
}
exports.lookForIdParameterAndReturnItsValue = lookForIdParameterAndReturnItsValue;
/**
 * Transform camelCase or snake_case to dashed-case,
 * so the routerboard can understand the parameters used
 * on this wrapper
 *
 * @param val to string to transform
 */
function camelCaseOrSnakeCaseToDashedCase(val) {
    // Clean any empty space left
    return val.replace(/ /g, "")
        // Convert camelCase to dashed
        .replace(/([a-z][A-Z])/g, (g, w) => {
        return g[0] + "-" + g[1].toLowerCase();
    })
        // Replace any underline to hiphen if used
        .replace(/_/g, "-");
}
exports.camelCaseOrSnakeCaseToDashedCase = camelCaseOrSnakeCaseToDashedCase;
/**
 * Transform routerboard's dashed-case to camelCase
 * so we can use objects properties without having to wrap
 * around quotes
 *
 * @param val the string to transform
 */
function dashedCaseToCamelCase(val) {
    return val.replace(/-([a-z])/g, (g) => {
        return g[1].toUpperCase();
    });
}
exports.dashedCaseToCamelCase = dashedCaseToCamelCase;
/**
 * Transform routerboard's dashed-case to snake_case
 * so we can use objects properties without having to wrap
 * around quotes
 *
 * @param val the string to transform
 */
function dashedCaseToSnakeCase(val) {
    return val.replace(/-/g, "_");
}
exports.dashedCaseToSnakeCase = dashedCaseToSnakeCase;
